### PR TITLE
Add score based example

### PR DESCRIPTION
**Pull request by lucekdudek**
Example output

```bash
golem-core-python/examples/score_based_providers$ python score_based_providers.py 
Scoring 0.7 proposal from 1nxt4s2.h - 0x74835ba53e9bf894ab06d4ec9cc557d5fa8ecdbb
Scoring 0.7 proposal from bristlecone.h - 0xccb9504c3a8fdb3a6984eb514112676829aa4202
Scoring None proposal from sycamore.h - 0xd385526f0a2d9165ba9aa704f4b9f33e85f7e8c2
Scoring 0.7 proposal from 10hx4r2.h - 0x87af9bf7c7e8f37ef091b86d245c119fe5acabba
...
Scoring 0.9 proposal from golem2004.h - 0xa569c2fc7757a0235bb02de7897a715536ebc698
Scoring 0.7 proposal from 10jx4r2.h - 0x85f85b8c071a5b2fea0eee7e0c824ace51ba469f
Scoring 0.9 proposal from golem2005.h - 0x52b10b8d1e7dc4891a435b932f2b6992c899d18a
Scoring None proposal from zoidberg - 0xe1ba9248dd9f6631fa09ee65b53a669774d8fdb1
Yielding proposal with score 0.9
Exception in default_negotiate(Proposal(R-60bb1f4dd8a1ace20e2d9637dbd7ec8b0a8ee3ebb9119439a22ad8ec72015033)): Negotiations based on Proposal(R-60bb1f4dd8a1ace20e2d9637dbd7ec8b0a8ee3ebb9119439a22ad8ec72015033) failed
Yielding proposal with score 0.9
Exception in default_negotiate(Proposal(R-da7ab3a55e943139e3bc6df422f3497bd4de5767bcc5ae4833898b607e3a36fe)): Negotiations based on Proposal(R-da7ab3a55e943139e3bc6df422f3497bd4de5767bcc5ae4833898b607e3a36fe) failed
Yielding proposal with score 0.8
RESULT: Hello from Activity(b390ef9436fe475e9c41e9e03c28b6d0)
TASK DONE
Scoring 0.8 proposal from imapp2030.h - 0x7ea10e5b7f76f973c753dc63928f1365f48a63c5
```

### CODE REVIEW

File: golem_core/mid/simple_scorer.py [hunk 1]
```diff
@@ -81,8 +87,11 @@ async def _wait_until_ready(self) -> None:
         start = datetime.now()
 
         while True:
+            await asyncio.sleep(0.1)
```
Comment (approxit) [inline golem_core/mid/simple_scorer.py:90]: Moving sleep at the beginning looks like not a good idea for me. It introduces unwanted delay, if work is node sooner.

File: golem_core/mid/simple_scorer.py [hunk 2]
```diff
@@ -81,8 +87,11 @@ async def _wait_until_ready(self) -> None:
...
+            if self._min_wait is not None and  datetime.now() - start < self._min_wait:
+                # force wait until time exceeds `min_wait`
+                continue
             if self._min_proposals is None or len(self._scored_proposals) >= self._min_proposals:
                 break
             if self._max_wait is not None and datetime.now() - start >= self._max_wait:
```
Comment (approxit) [inline golem_core/mid/simple_scorer.py:96]: As `datetime.now()` is called twice, introduce variable to cache its value, or even result of `datetime.now() - start`.

File: pyproject.toml [hunk 3]
```diff
@@ -53,7 +53,7 @@ implicit_reexport = true
 allow_untyped_calls = true
 
 [tool.poe.tasks]
-test = "pytest"
+test = { cmd = "pytest", env = { YAGNA_SUBNET = "public" } }
```
Comment (approxit) [inline pyproject.toml:56]: Why this change is required as there is no tests introduced in this PR?

Comment (lucekdudek) [reply pyproject.toml:56]: Old test were failing as there are no providers on default subnet in yappai version used by golem-core

File: golem_core/mid/simple_scorer.py (hunk 1 above)
Comment (lucekdudek) [reply golem_core/mid/simple_scorer.py:90]: `sleep` has to be before `continue` block otherwise it will deadlock all other asyncio tasks.

File: golem_core/mid/simple_scorer.py (hunk 2 above)
Comment (lucekdudek) [reply golem_core/mid/simple_scorer.py]: Added

**Review by approxit — APPROVED** [review]

**Review by shadeofblue — APPROVED** [review]
lgtm
